### PR TITLE
Damoyolo test process done.

### PR DESCRIPTION
**Pull request by ueanperfect**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/damoyolo/damo_yolo_s.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+_base_ = '../_base_/default_runtime.py'
+
+# dataset settings
+data_root = 'data/coco25/'
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:4]: 名字改成标准的

File: configs/damoyolo/damo_yolo_s.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+dataset_type = 'mmdet.CocoDataset'
+
+# parameters that often need to be modified
+img_scale = (640, 640)  # height, width
+max_epochs = 300
+save_epoch_intervals = 10
+train_batch_size_per_gpu = 2
+train_num_workers = 2
+val_batch_size_per_gpu = 25
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:13]:
```suggestion
val_batch_size_per_gpu = 1
```

File: configs/damoyolo/damo_yolo_s.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+val_num_workers = 2
+base_lr = 0.01
+
+# persistent_workers must be False if num_workers is 0.
+persistent_workers = True
+
+model = dict(
+    type='DAMOYOLODetector',
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:21]: 不需要改这个

File: configs/damoyolo/damo_yolo_s.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        lr=base_lr,
+        momentum=0.937,
+        weight_decay=0.0005,
+        nesterov=True,
+        batch_size_per_gpu=train_batch_size_per_gpu),
+    constructor='YOLOv5OptimizerConstructor')
+
+val_dataloader = dict(
+    batch_size=val_batch_size_per_gpu,
+    num_workers=2,
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:133]: 用前面的参数

File: configs/damoyolo/damo_yolo_s.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    persistent_workers=True,
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:134]: 用参数，不要写死

File: configs/damoyolo/damo_yolo_s.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    pin_memory=True,
+    drop_last=False,
+    sampler=dict(type='DefaultSampler', shuffle=False),
+    dataset=dict(
+        type='YOLOv5CocoDataset',
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:139]: 用前面的参数

File: configs/damoyolo/damo_yolo_s.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                 height=img_scale[0],
+                 keep_ratio=True,
+                 interpolation='bilinear'),
+            dict(type='LoadAnnotations', with_bbox=True, _scope_='mmdet'),
+            dict(
+                type='mmdet.PackDetInputs',
+                meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
+                           'scale_factor'))
+        ],
+        batch_shapes_cfg=dict(
```
Comment (hhaAndroid) [inline configs/damoyolo/damo_yolo_s.py:160]: 官方用了这个吗？

File: tools/misc/extract_subcoco.py [hunk 8]
```diff
@@ -157,4 +157,4 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (hhaAndroid) [inline tools/misc/extract_subcoco.py:160]: 不要改

File: tools/test.py [hunk 9]
```diff
@@ -122,7 +122,8 @@ def main():
         runner.test_evaluator.metrics.append(
             DumpResults(out_file_path=args.out))
 
-    # start testing
+    # import torch
+    # torch.save(runner.model.state_dict(),'damoyolo.pth')
```
Comment (hhaAndroid) [inline tools/test.py:126]: 不要改

File: tools/train.py [hunk 10]
```diff
@@ -103,4 +103,4 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (hhaAndroid) [inline tools/train.py:106]: 不要改

File: model_sructure [hunk 11]
```diff
@@ -0,0 +1,16 @@
+SuperResStem(
```
Comment (hhaAndroid) [inline model_sructure:1]: 这个文件不要上传

File: mmyolo/models/detectors/damoyolo_detector.py [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+from mmengine.logging import print_log
+from ..backbones.tinynas_res import RepConv
+import torch.nn as nn
+from mmyolo.registry import MODELS
+from mmdet.structures import  SampleList
+from torch import Tensor
+
+
+@MODELS.register_module()
+class DAMOYOLODetector(SingleStageDetector):
```
Comment (hhaAndroid) [inline mmyolo/models/detectors/damoyolo_detector.py:15]: 不要重写

File: configs/damoyolo/damo_yolo_s.py (hunk 2 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py]: 已经修改过来了

File: configs/damoyolo/damo_yolo_s.py (hunk 3 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py:21]: 我思考了一下，那个合并层部分可能还是需要在detector里面写会比较好，暂时按照这样来吧。

File: configs/damoyolo/damo_yolo_s.py (hunk 4 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py]: 已经修改过来了。

File: configs/damoyolo/damo_yolo_s.py (hunk 5 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py]: 已经修改过来了。

File: configs/damoyolo/damo_yolo_s.py (hunk 7 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py:160]: 官方是按照这个策略的。

File: model_sructure (hunk 11 above)
Comment (ueanperfect) [reply model_sructure]: 已经删除了

File: mmyolo/models/detectors/damoyolo_detector.py (hunk 12 above)
Comment (ueanperfect) [reply mmyolo/models/detectors/damoyolo_detector.py:15]: 上述已经说明情况了。

File: tools/test.py (hunk 9 above)
Comment (ueanperfect) [reply tools/test.py]: 已经修改过来了

File: tools/misc/extract_subcoco.py (hunk 8 above)
Comment (ueanperfect) [reply tools/misc/extract_subcoco.py]: 已经还原了

File: configs/damoyolo/damo_yolo_s.py (hunk 3 above)
Comment (ueanperfect) [reply configs/damoyolo/damo_yolo_s.py:21]: 主要是推理和训练的部分，模型结构不一样，如果按照原来的那个想法来，要不先按照这个来，之后再看如何选择比较好。我觉得这个设计模型结构方面的讨论。

**Pull request merged**